config/templates/publish_mdm/devices_list.html

```diff
@@ -1,73 +1,116 @@
+                            <path stroke="currentColor" stroke-linejoin="round" stroke-width="2" d="M7 7h.01v.01H7V7Zm10 10h.01v.01H17V17Z" />
+                        </svg>
+                        Enroll
+                    </button>
+                    <div class="flex items-center w-full md:w-auto">
+                        <button id="actionsDropdownButton"
+                                data-dropdown-toggle="actionsDropdown"
+                                class="w-full md:w-auto flex items-center justify-center py-2 px-4 text-sm font-medium text-gray-900 focus:outline-none bg-white rounded-lg border border-gray-200 hover:bg-gray-100 hover:text-primary-700 focus:z-10 focus:ring-4 focus:ring-gray-200 dark:focus:ring-gray-700 dark:bg-gray-800 dark:text-gray-400 dark:border-gray-600 dark:hover:text-white dark:hover:bg-gray-700"
+                                type="button">
+                            <svg class="-ml-1 mr-1.5 w-5 h-5"
+                                 fill="currentColor"
+                                 viewbox="0 0 20 20"
                                  xmlns="http://www.w3.org/2000/svg"
-                                 fill="none"
-                                 viewBox="0 0 14 14">
-                                <path stroke="currentColor" stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="m1 1 6 6m0 0 6 6M7 7l6-6M7 7l-6 6" />
+                                 aria-hidden="true">
+                                <path clip-rule="evenodd" fill-rule="evenodd" d="M5.293 7.293a1 1 0 011.414 0L10 10.586l3.293-3.293a1 1 0 111.414 1.414l-4 4a1 1 0 01-1.414 0l-4-4a1 1 0 010-1.414z" />
                             </svg>
-                            <span class="sr-only">Close modal</span>
+                            Actions
                         </button>
+                        <div id="actionsDropdown"
+                             class="hidden z-100 w-44 bg-white rounded divide-gray-100 shadow dark:bg-gray-700 dark:divide-gray-600">
+                            <ul class="py-1 text-sm text-gray-700 dark:text-gray-200"
+                                aria-labelledby="actionsDropdownButton">
+                                <li>
+                                    <button name="sync"
+                                            hx-post="{{ request.get_full_path }}"
+                                            hx-target="#inner-table"
+                                            hx-swap="outerHTML"
+                                            hx-push-url="true"
+                                            hx-indicator=".syncing"
+                                            hx-disabled-elt="this"
+                                            class="w-full text-left py-2 px-4 hover:bg-gray-100 dark:hover:bg-gray-600 dark:hover:text-white disabled:opacity-75 disabled:cursor-wait">
+                                        Sync From MDM
```

Comment:
One thought: If this wasn't a dropdown, we could use a similar loading treatment like we do on the publish page here:

https://github.com/user-attachments/assets/bf6ff8a1-f469-49bc-92a3-48c8c5e4d89f

Not sure if that's harder to do or not, but it might make it look a bit cleaner.